### PR TITLE
Minor fixes to proximity sensors

### PR DESCRIPTION
**Pull request by ZomgPonies**
- Ports https://github.com/PolarisSS13/Polaris/pull/375

### CODE REVIEW

File: code/game/turfs/turf.dm [hunk 1]
```diff
@@ -108,7 +108,7 @@
 //		O.update_gravity(O.mob_has_gravity(src))
 
 	var/loopsanity = 100
-	for(var/atom/A in range(1))
+	for(var/atom/movable/A in range(1))
```
Comment (tigercat2000) [inline code/game/turfs/turf.dm]: this is still an atom proc and therefore should use the atom loop

**Comment by Spacemanspark** [comment]
Next, you should port Communicators from Polaris.
:^)

**Comment by tigercat2000** [comment]
we were actually just talking about that @spacemanspark

**Comment by Spacemanspark** [comment]
Wait really?
Fuck yes, those are fun.

File: code/game/turfs/turf.dm (hunk 1 above)
Comment (ZomgPonies) [inline code/game/turfs/turf.dm]: This is how everyone else has it configured. It's not supposed to pass any turfs or areas.

Comment (tigercat2000) [inline code/game/turfs/turf.dm]: why is it an *atom* proc then
not that it has any use whatsoever

Comment (ZomgPonies) [inline code/game/turfs/turf.dm]: Beats me! Leaving it as is for consistency with other lesser codebases.

Comment (Fox-McCloud) [inline code/game/turfs/turf.dm]: This isn't how everyone else has it configured--as @tigercat2000 pointed out, it's an atom proc, and tehrefore checks for atoms: https://github.com/tgstation/-tg-station/blob/ba418abbbf9a086065f370d2f9957dbc54164fc2/code/game/turfs/turf.dm#L96

Comment (ZomgPonies) [inline code/game/turfs/turf.dm]: https://github.com/tgstation/-tg-station/blob/ba418abbbf9a086065f370d2f9957dbc54164fc2/code/game/atoms.dm#L107

And then casts them as atom/movable.

Comment (tigercat2000) [inline code/game/turfs/turf.dm]: ... um, what

It's an /atom/proc
which takes an argument of an /atom/movable
this is correct, the for loop is *calling* HasProximity on every atom nearby, with the argument of what is proximate to them

Comment (ZomgPonies) [inline code/game/turfs/turf.dm]: @Zuhayr HALP

Comment (Zuhayr) [inline code/game/turfs/turf.dm]: This was an emergency fix for server-crashing runtime spam so I didn't integrate it as thoroughly as I would have liked. Tiger has a point about changing it to atom/movable, it does fix the problem but it also means that it won't be possible for turfs to use HasProximity in the future. Ultimately it's not particularly relevant as no turfs currently use it as far as I know, but the point stands.

**Pull request closed without merging**